Library/Homebrew/cmd/update-report.rb

```diff
@@ -249,8 +249,25 @@ def updated?
   def migrate_tap_migration
     report[:D].each do |full_name|
       name = full_name.split("/").last
+      new_tap_name = tap.tap_migrations[name]
+      next if new_tap_name.nil? # skip if not in tap_migrations list.
+
+      if tap == "caskroom/cask"
+        next unless (HOMEBREW_REPOSITORY/"Caskroom"/name).exist?
+        new_tap = Tap.fetch(new_tap_name)
+        new_tap.install unless new_tap.installed?
+        ohai "#{name} has been moved to Homebrew. Installing #{name}...", <<-EOS.undent
+          To uninstall the cask run:
+            brew cask uninstall --force #{name}
+        EOS
+        system HOMEBREW_BREW_FILE, "install", "#{new_tap_name}/#{name}"
+        unless Formulary.factory("#{new_tap_name}/#{name}").keg_only?
```

Comment:
Loading formulae need to be guarded by `begin ... rescue Execption ... end`. You can see the example codes in this very file at below. 

Also I think it's worth to check whether this formula is installed before to install it.